### PR TITLE
Only release contracts with Proxies

### LINKED ISSUE #5474: `make-release` only releases contracts with proxies

### PR DESCRIPTION
**Pull request by nambrot**
### Description

This adds an additional check to only deploy contracts (or consider deploying) that have an accompanying proxy contract

### Tested

- Dry-ran it on Alfajores

### Related issues

- Fixes #5474 

### Backwards compatibility

Should be

### CODE REVIEW

File: packages/protocol/scripts/truffle/make-release.ts [hunk 1]
```diff
@@ -72,6 +72,19 @@ class ContractAddresses {
 
 const REGISTRY_ADDRESS = '0x000000000000000000000000000000000000ce10'
 
+const isCoreContract = async (contractName: string) => {
```
Comment (m-chrzan) [inline packages/protocol/scripts/truffle/make-release.ts]: Would it be better to call this `isProxiedContracts`? Is it the case that there are no (and there won't be) contracts with a corresponding Proxy that wouldn't be considered core Celo contracts? 

Would using `CeloContractName` from `lib/registry-utils.ts` be more appropriate?

File: packages/protocol/scripts/truffle/make-release.ts [hunk 2]
```diff
@@ -210,7 +223,9 @@ module.exports = async (callback: (error?: any) => number) => {
       }
     }
     for (const contractName of contracts) {
-      await release(contractName)
+      if (await isCoreContract(contractName)) {
```
Comment (m-chrzan) [inline packages/protocol/scripts/truffle/make-release.ts]: Will this not skip all linked libraries?

File: packages/protocol/scripts/truffle/make-release.ts (hunk 1 above)
Comment (nambrot) [reply packages/protocol/scripts/truffle/make-release.ts]: I did both!

File: packages/protocol/scripts/truffle/make-release.ts (hunk 2 above)
Comment (nambrot) [reply packages/protocol/scripts/truffle/make-release.ts]: I believe each `release` will deploy the linked library separately

File: packages/protocol/scripts/truffle/make-release.ts [hunk 3]
```diff
@@ -72,6 +73,22 @@ class ContractAddresses {
 
 const REGISTRY_ADDRESS = '0x000000000000000000000000000000000000ce10'
 
+const isProxiedContract = async (contractName: string) => {
+  if (contractName.endsWith('Proxy')) {
+    return false
+  }
+
+  try {
+    await artifacts.require(`${contractName}Proxy`)
```
Comment (m-chrzan) [inline packages/protocol/scripts/truffle/make-release.ts]: I don't think you need to `await` on `artifacts.require`, so this whole function doesn't need to be async.

**Review by m-chrzan — APPROVED** [review]

**Pull request merged**